java/memory/memory-core/src/main/java/org/apache/arrow/memory/LowCostIdentityHashMap.java

```diff
@@ -38,18 +38,18 @@ public class LowCostIdentityHashMap<K, V extends ValueWithKeyIncluded<K>> {
   private @Nullable Object[] elementData; // elementData[index] = null;
 
   /* Actual number of values. */
-  private int size;
```

Comment:
I guess we can leave this hash class with the changes?

Reply:
We can leave it.

This is _not_ about replacing int with long. We only want to replace things where the `int` is being used as an address or offset.

Reply:
Sure. I will revert.